### PR TITLE
[Docs] Add WIP disclaimer

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
State in README and web docs intro that this project is a WIP. To be removed, when we're ready for 1.0.0 relase.

Readme preview: https://github.com/lukaszstolarczuk/unified-memory-framework/tree/wip-disclaimer?tab=readme-ov-file#introduction
Web preview: https://lukaszstolarczuk.github.io/unified-memory-framework/introduction.html

### CODE REVIEW

**Review by bratpiorka — APPROVED** [review]

**Comment by rozhukov** [comment]
Everything looks good to me. I'd add 1 more foil:
...that the Software:
- ...
- is not eligible for Intel® Bug Bounty Program

File: README.md [hunk 1]
```diff
@@ -15,6 +15,17 @@
 
 The Unified Memory Framework (UMF) is a library for constructing allocators and memory pools. It also contains broadly useful abstractions and utilities for memory management. UMF allows users to manage multiple memory pools characterized by different attributes, allowing certain allocation types to be isolated from others and allocated using different hardware resources as required.
 
+**⚠️ Work-In-Progress Disclaimer:** \
+Recipient understands, acknowledges and agrees that the Software:
+> - is a pre-production, research version intended for use in a testing simulator;
```
Comment (bratpiorka) [inline README.md:20]: "simulator -> environment" would be better?

Comment (lukaszstolarczuk) [reply README.md]: I re-phrased the first sentence.

**Comment by lukaszstolarczuk** [comment]
Re-phrased it.

File: README.md [hunk 2]
```diff
@@ -15,6 +15,12 @@
 
 The Unified Memory Framework (UMF) is a library for constructing allocators and memory pools. It also contains broadly useful abstractions and utilities for memory management. UMF allows users to manage multiple memory pools characterized by different attributes, allowing certain allocation types to be isolated from others and allocated using different hardware resources as required.
 
+**⚠️ Work-In-Progress disclaimer:** <br />
+> Please note that this project is **pre-production software**, it should not be considered complete or fully functional.
+> It has not been fully tested yet (including security testing). This means it is not eligible for Intel® Bug Bounty Program.
```
Comment (rozhukov) [inline README.md:20]: Replace:
> It has not been fully tested yet (including security testing). This means it is not eligible for Intel® Bug Bounty Program.

By:
It has not been fully tested yet (including security testing). It is not recommended to be used in production as part of a larger system. It is not eligible for Intel® Bug Bounty Program.

Comment (lukaszstolarczuk) [reply README.md]: Done.

**Review by lplewa — APPROVED** [review]
remove: <br />

**Pull request merged**